rir/src/interpreter/interp.c

```diff
@@ -494,27 +480,22 @@ INLINE SEXP rirCallClosure(SEXP call, SEXP env, SEXP callee, SEXP actuals,
 
     ostack_push(ctx, newEnv);
 
-    // Create a new R context and store it on the stack
```

Comment:
So at some point the context of a function was allocated on the heap. I can't remember the exact details. To be able to put it on the interpreter stack it was boxed in a RAWSXP. This changes it to allocate the context on the C runtime stack. Since the execution of the R function coincides with the execution of the rirCallClosure C function this works well.

For example in the case of the loop context that does not work and we still allocate loop contexts on the R heap and store it in the RIR interpreter stack.